llvm/lib/Target/SPIRV/SPIRVBuiltins.h

```diff
@@ -27,13 +28,14 @@ namespace llvm {
 /// \p DemangledCall is the skeleton of the lowered builtin function call.
 /// \p Set is the external instruction set containing the given builtin.
 /// \p OrigRet is the single original virtual return register if defined,
-/// Register(0) otherwise. 
-/// \p OrigRetTy is the type of the \p OrigRet. 
+/// Register(0) otherwise.
+/// \p OrigRetTy is the type of the \p OrigRet.
 /// \p Args are the arguments of the lowered builtin call.
-std::pair<bool, bool> lowerBuiltin(
-    const StringRef DemangledCall, SPIRV::InstructionSet::InstructionSet Set,
-    MachineIRBuilder &MIRBuilder, const Register OrigRet, const Type *OrigRetTy,
-    const SmallVectorImpl<Register> &Args, SPIRVGlobalRegistry *GR);
+std::pair<bool, bool>
```

Comment:
Getting back to my original concerns about this, I though that `std::optional<bool>` should be a perfect replacement for this pair. As LLVM currently support C++17 by default, we're ok from that side too.

Reply:
BTW I became curious to try that myself, so https://reviews.llvm.org/D132802

Reply:
Isn't it better use `llvm::Optional` (llvm/ADT/Optional.h)? It's widely used in llvm. On other hand, `std::optional` is not used at all in llvm/lib.